### PR TITLE
[WFLY-7764] - Fixes to get JACC TCK tests passing

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Review by darranl — CHANGES_REQUESTED** [review]
Main thing missing is we will need a transformer for the new attribute in the security subsystem, if set to anything other than the default value we need to reject it.

File: ejb3/src/main/java/org/jboss/as/ejb3/component/EJBComponentDescription.java [hunk 1]
```diff
@@ -682,12 +682,22 @@ public void setDefaultSecurityDomain(final String defaultSecurityDomain) {
         this.defaultSecurityDomain = defaultSecurityDomain;
     }
 
-    public void setKnownSecurityDomainPredicate(final Predicate<String> knownSecurityDomain) {
+    public void setKnownSecurityDomainPredicate(final Function<String, ApplicationSecurityDomainConfig> knownSecurityDomain) {
```
Comment (fjuma) [inline ejb3/src/main/java/org/jboss/as/ejb3/component/EJBComponentDescription.java]: Just a minor nit, this method should probably be renamed to "setKnownSecurityDomainFunction" now.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/ApplicationSecurityDomainDefinition.java [hunk 2]
```diff
@@ -179,8 +206,8 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
         }
     }
 
-    Predicate<String> getKnownSecurityDomainPredicate() {
-        return knownApplicationSecurityDomains::contains;
+    Function<String, ApplicationSecurityDomainConfig> getKnownSecurityDomainPredicate() {
```
Comment (fjuma) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/ApplicationSecurityDomainDefinition.java]: Same minor nit here, should probably rename this to "getKnownSecurityDomainFunction" now.

File: ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class JaccInterceptor implements Interceptor {
+
+    private final String contextID;
+    private final String viewClassName;
+    private final Method viewMethod;
+
+    public JaccInterceptor(String contextID, String viewClassName, Method viewMethod) {
+        this.contextID = contextID;
```
Comment (fjuma) [inline ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java]: One thing to note is that the EJB subsystem currently contains another interceptor that also sets the policy context id (org.jboss.as.ejb3.security.PolicyContextIdInterceptor). May be able to remove that one now.

Comment (pedroigor) [reply ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java]: Yes. Let me check if we can remove it.

Comment (pedroigor) [reply ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java]: @fjuma We need that interceptor, but it seems we can remove some code from the new ```JaccInterceptor``` and avoid setting the context id twice. I'm going to perform some tests and see if that is OK.

**Comment by stuartwdouglas** [comment]
Just FYI JACC does already pass (due to https://github.com/wildfly-security-incubator/wildfly/commit/803ec8c719a7753303afbb85ae31f1b0464221f1 which this is conflicting with), although the changes in the interceptor look like they will be necessary to get role mapping to work.

**Comment by darranl** [comment]
I have merged the interceptor commit so now going to close this PR as the conflict is duplication.

**Pull request closed without merging**